diesel_codegen/src/associations.rs

```diff
@@ -0,0 +1,103 @@
+use syn;
+use quote;
+
+use model::{Model, infer_association_name};
+use util::str_value_of_meta_item;
+
+pub fn derive_associations(input: syn::MacroInput) -> quote::Tokens {
+    let mut derived_associations = Vec::new();
+    let model = t!(Model::from_item(&input, "Associations"));
```

Comment:
Instead of using strings here (and const lists of strings in lib), you could use a set of modules with constants for derives and attributes [like this](https://play.rust-lang.org/?gist=6a7fea0458a4d3fbd8cbf9b3fbc5c513&version=stable&backtrace=0). (Doesn't need to be in this PR. I just hate doing stringly typed stuff in a language like Rust 😉 )